### ISSUE TITLE
Code breaks when moving from Linux to MacOS

### ISSUE THREAD

**Issue by jstovold** [danielegrattarola/spektral#443]
I have a piece of Spektral-based code for which I am using my university HPC cluster. Everything appears to work fine on the cluster (in terms of errors etc. anyway). I just downloaded the notebook to run on my local machine while I do not have an internet connection, and I now get the following error:

```ValueError: Unsupported type <class 'scipy.sparse._coo.coo_array'> for a```

I have checked that both systems are running the same version of Spektral (1.3.0), so I don't see why the code works fine on one system and not the other. The error is triggered by a simple call to the Graph constructor in the Dataset.read() method, like:

```return [Graph(x=self.all_x[b,...], a = self.a) for b in self.currentbatch]```

Any ideas on how I can diagnose this problem? Is there some underlying dependency that I need to check the version for?

**Comment by jstovold** [comment]
I've managed to fix the problem in my code, but there might still be an underlying issue in Spektral relating to it. I fixed it by creating the adjacency matrix as a coo_matrix instead of  a coo_array.

**Comment by danielegrattarola** [comment]
Yeah I haven't seen that coo_array type before and Spektral assumes a COO matrix, but I'll look into extending support. 
Thanks for raising this
